Sources/GRPCHTTP2Core/GRPCStreamStateMachine.swift

```diff
@@ -1469,6 +1521,43 @@ extension GRPCStreamStateMachine {
       return .awaitMoreMessages
     }
   }
+
+  private func getServerUnexpectedInboundCloseAction(
+    for reason: UnexpectedInboundCloseReason
+  ) -> OnUnexpectedInboundClose {
+    switch reason {
+    case .streamReset, .channelInactive:
+      // We can force-unwrap here: the status code is `.unavailable`, so RPCError won't be nil
+      return .fireError(RPCError(status: Status(reason))!)
```

Comment:
Does this hint at the wrong construction? I can't imagine an unexpected close would ever result in an `ok` status, so it seems like the `RPCError` should be created from the reason and then, where necessary, a `Status` from the `RPCError`. WDYT?